python/ray/serve/api.py

```diff
@@ -571,3 +572,97 @@ def delete(name: str, _blocking: bool = True):
+    necessary.
+
+    When the number of models in one replica is larger than max_num_models_per_replica,
+    the models will be unloaded using an LRU policy.
+
+    If you want to release resources after the model is loaded, you can define
+    a `__del__` method in your model class. The `__del__` method will be called when
+    the model is unloaded.
+
+    Example:
+
+    .. code-block:: python
+            from ray import serve
+
+            @serve.deployment
+            class MultiplexedDeployment:
+
+                def __init__(self):
+                    # Define s3 base path to load models.
+                    self.s3_base_path = "s3://my_bucket/my_models"
+
+                @serve.multiplexed(max_num_models_per_replica=5)
+                async def load_model(self, model_id: str) -> Any:
+                    # Load model with the given tag
+                    # You can use any model loading library here
+                    # and return the loaded model. load_from_s3 is
+                    # a placeholder function.
+                    return load_from_s3(model_id)
+
+                async def __call__(self, request):
+                    # Get the model_id from the request context.
+                    model_id = serve.get_multiplexed_model_id()
+                    # Load the model for the requested model_id.
+                    # If the model is already cached locally,
+                    # this will just be a dictionary lookup.
+                    model = await self.load_model(model_id)
+                    return model(request)
+
+
+    Args:
+        max_num_models_per_replica: the maximum number of models
+        to be loaded on each replica. By default, it is 3, which
+        means that each replica can cache up to 3 models. You can
+        set it to a larger number if you have enough memory on
+        the node resource, in opposite, you can set it to a smaller
+        number if you want to save memory on the node resource.
+    """
+
+    raise NotImplementedError("Multiplexed deployment is not supported yet.")
+
+
+@PublicAPI(stability="alpha")
+def get_multiplexed_model_id() -> str:
+    """[EXPERIMENTAL] Get the multiplexed model ID for the current request.
+
+    This is used with a function decorated with `@serve.multiplexed`
+    to cache multiple independent models within a single deployment.
+
+    .. code-block:: python
+            from ray import serve
+            import ray
+            import requests
+            # client code
+            # User needs to set the model id in the request header with the key
+            # "ray_serve_request_model_id" when sending requests to the http proxy.
+            requests.get("http://localhost:8000",
+                headers={"ray_serve_request_model_id": "model_1"})
+            # Or user can also set model id inside the serve handle.
+            handle.options(model_id="model_1").remote("blablabla")
```

Comment:
```suggestion
            requests.get("http://localhost:8000",
                headers={"ray_serve_multiplexed_model_id": "model_1"})
            # Or user can also set model id inside the serve handle.
            handle.options(multiplexed_model_id="model_1").remote("blablabla")
```
let's please keep the naming consistent across the API

Reply:
we might not need the `ray_serve_` prefix in the header, i don't have too much preference

Reply:
I think it is more readable to the user for setting up the headers. i will keep it as it is right now.